internal/autoevent/executor.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2019 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package autoevent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/handler"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+const millionSecPerDay uint32 = 86400000
+
+type Executor interface {
+	Run()
+	Stop()
+}
+
+type executor struct {
+	deviceName   string
+	autoEvent    models.AutoEvent
+	lastReadings map[string]string
+	duration     time.Duration
+	stop         bool
+}
+
+func (e *executor) Run() {
+	for {
+		if e.stop {
+			break
+		}
+		time.Sleep(e.duration)
+
+		common.LoggingClient.Debug(fmt.Sprintf("AutoEvent - executing %v", e.autoEvent))
+		evt, appErr := readResource(e)
+		if appErr != nil {
+			common.LoggingClient.Error(fmt.Sprintf("AutoEvent - error occurs when read resource %s",
+				e.autoEvent.Resource))
+			continue
+		} else if evt != nil {
```

Comment:
Is there a scenario where evt and appErr can both be nil?

Reply:
No, I don't think so.  Should we add this check?

Reply:
For starters, the go style isn't to chain if/else code like this. Instead it should be:
```

if appErr != nil {
}

if evt != nil {
}

if evt == nil {
}

// handle evt
```

Reply:
modified